test/integration/controller/jobs/xgboostjob/xgboostjob_controller_test.go

```diff
@@ -0,0 +1,575 @@
+
+		ginkgo.By("checking the job is unsuspended when workload is assigned")
+		onDemandFlavor := testing.MakeResourceFlavor("on-demand").Label(labelKey, "on-demand").Obj()
+		gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).Should(gomega.Succeed())
+		spotFlavor := testing.MakeResourceFlavor("spot").Label(labelKey, "spot").Obj()
+		gomega.Expect(k8sClient.Create(ctx, spotFlavor)).Should(gomega.Succeed())
+		clusterQueue := testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(
+				*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "5").Obj(),
+				*testing.MakeFlavorQuotas("spot").Resource(corev1.ResourceCPU, "5").Obj(),
+			).Obj()
+		admission := testing.MakeAdmission(clusterQueue.Name).
+			PodSets(
+				kueue.PodSetAssignment{
+					Name: "Master",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "on-demand",
+					},
+					Count: ptr.To(createdWorkload.Spec.PodSets[0].Count),
+				},
+				kueue.PodSetAssignment{
+					Name: "Worker",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "spot",
+					},
+					Count: ptr.To(createdWorkload.Spec.PodSets[1].Count),
+				},
+			).
+			Obj()
+		gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+		util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+		lookupKey := types.NamespacedName{Name: jobName, Namespace: ns.Name}
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+				return false
+			}
+			return !*createdJob.Spec.RunPolicy.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		gomega.Eventually(func() bool {
+			ok, _ := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
+			return ok
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		gomega.Expect(len(createdJob.Spec.XGBReplicaSpecs[kftraining.XGBoostJobReplicaTypeMaster].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJob.Spec.XGBReplicaSpecs[kftraining.XGBoostJobReplicaTypeMaster].Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(onDemandFlavor.Name))
+		gomega.Expect(len(createdJob.Spec.XGBReplicaSpecs[kftraining.XGBoostJobReplicaTypeWorker].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJob.Spec.XGBReplicaSpecs[kftraining.XGBoostJobReplicaTypeWorker].Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(spotFlavor.Name))
+		gomega.Consistently(func() bool {
+			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+				return false
+			}
+			return len(createdWorkload.Status.Conditions) == 2
+		}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+
+		ginkgo.By("checking the job gets suspended when parallelism changes and the added node selectors are removed")
```

Comment:
non-blocking nit: we could make the integration test shorter as the implementation is pretty well commonized. I'm thinking of this scenario as a potential overkill, but I don't have a strong view as some integration tests are needed as a scenarios like this is probably <1s anyway.

Reply:
That's a good point.

> we could make the integration test shorter as the implementation is pretty well commonized. I'm thinking of this scenario as a potential overkill

Which does that indicate all integrations or all kubeflow jobs?

Reply:
We have pretty well commonized code at both levels - between integrations, and kubeflow integrations in particular.  As a result each kubeflow integration implements around 10 functions, and all are one-liners.

Reply:
> That's a good point.
> 
> > we could make the integration test shorter as the implementation is pretty well commonized. I'm thinking of this scenario as a potential overkill
> 
> Which does that indicate all integrations or all kubeflow jobs?

I would focus only on kubeflow jobs here.

Reply:
It makes sense. I think moving this case to a unit test might be better since this scenario might be useful to check expected behavior although this is overkill as integration tests.
WDYT?

Reply:
sgtm, still I would kieep some basic integration tests per integratoin, can be follow up

Reply:
Let's keep tracking by creating an issue.

Reply:
sgtm, would you like to create one?

Reply:
I apologize that I forgot to push `Submit new Issue` button 😞 
Created: https://github.com/kubernetes-sigs/kueue/issues/1119